packages/core/src/utilities/progressiveIterator.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * A progressive iterator is an async iterator that can have data delivered
```

Comment:
@sedghi  - could you take a look at this class?  It is fairly simple, but I plan on using it as a base for delivering multiple updates for different types of progressive display - and perhaps eventually for region display.  I added a test to show how it gets used.